<launch>
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="worlds/empty.world"/> <!-- Note: the world_name is with respect to GAZEBO_RESOURCE_PATH environmental variable -->
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="recording" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="track_description" command="$(find xacro)/xacro $(find auto_rc_car_worlds)/urdf/concave_track.xacro.urdf" />
  <node name="spawn_track_urdf" pkg="gazebo_ros" type="spawn_model" args="-param track_description -x 0.0 -y -3.0 -z 0.00 -Y 1.57 -urdf -model circle_track --verbose" respawn="false" output="screen"/>


  <include file="$(find auto_rc_car_worlds)/launch/load_stoplight.launch">
    <arg name="ns" value="sign0"/>
    <arg name="x" value="5.0"/>
    <arg name="y" value="-2.0"/>
    <arg name="yaw" value="3.1415"/>
  </include>

  <include file="$(find auto_rc_car_worlds)/launch/load_stoplight.launch">
    <arg name="ns" value="sign1"/>
    <arg name="x" value="0"/>
    <arg name="y" value="-12.5"/>
    <arg name="yaw" value="0"/>
  </include>

  <include file="$(find auto_rc_car_worlds)/launch/load_stoplight.launch">
    <arg name="ns" value="sign2"/>
    <arg name="x" value="-18.5"/>
    <arg name="y" value="-0"/>
    <arg name="yaw" value="1.5707"/>
  </include>

  <include file="$(find auto_rc_car_worlds)/launch/load_pedestrian.launch">
    <arg name="ns" value="pedestrian0"/>
    <arg name="x" value="-17"/>
    <arg name="y" value="1"/>
    <arg name="yaw" value="3.1415"/>
  </include>
<!--

  
  <group ns="sign2">
    <param name="robot_description" command="$(find xacro)/xacro $(find auto_rc_car_worlds)/urdf/stoplight.xacro.urdf" />

    <node name="spawn_sign_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -x -18.5 -y -0.0 -z 0.00 -Y -1.5707 -urdf -model stoplight2 -verbose" respawn="false" output="screen"/>
    <rosparam  command="load" file="$(find auto_rc_car_worlds)/config/stoplight_controller.yaml"/>
    <node name="world_controller_spawner"
      pkg="controller_manager"
      type="spawner"
      output="screen"
      args="stoplight_controller">
    </node>
  </group>
  -->
</launch>